tests/jax/components/building/distributor_test.py

```diff
@@ -0,0 +1,265 @@
+@pytest.fixture
+def mock_builder() -> MockBuilder:
+    """Mock builder"""
+    return MockBuilder()
+
+
+def test_initiator() -> None:
+    """Test initiator of the Distributor"""
+    distributor = Distributor()
+    assert distributor.config.nodes_on_gpu == ["trainer"]
+
+
+def test_on_building_program_nodes_multi_process(mock_builder: MockBuilder) -> None:
+    """Test on_building_program_nodes for multi_process distributor
+
+    Args:
+        mock_builder: MockBuilder
+    """
+    distributor = Distributor()
+    distributor.on_building_program_nodes(builder=mock_builder)
+
+    assert isinstance(mock_builder.store.program, Launcher)
+
+    assert list(mock_builder.store.program._program._groups.keys()) == [
+        "data_server",
+        "parameter_server",
+        "executor",
+        "evaluator",
+        "trainer",
+    ]
+    assert (
+        mock_builder.store.program._program._groups["data_server"][
+            -1
+        ]._priority_tables_fn
+        == mock_builder.data_server
+    )
+    assert (
+        mock_builder.store.program._program._groups["parameter_server"][-1]._constructor
+        == mock_builder.parameter_server
+    )
+    assert (
+        mock_builder.store.program._program._groups["executor"][-1]._constructor
+        == mock_builder.executor
+    )
+    assert (
+        mock_builder.store.program._program._groups["evaluator"][-1]._constructor
+        == mock_builder.executor
+    )
+    assert (
+        mock_builder.store.program._program._groups["trainer"][-1]._constructor
+        == mock_builder.trainer
+    )
+
+    with pytest.raises(Exception):
+        mock_builder.store.program.get_nodes()
+
+
+def test_on_building_program_nodes_multi_process_no_evaluator(
+    mock_builder: MockBuilder,
+) -> None:
+    """Test on_building_program_nodes for multi_process distributor and no evaluator runs
+
+    Args:
+        mock_builder: MockBuilder
+    """
+    distributor = Distributor()
+    distributor.config.run_evaluator = False
+    distributor.on_building_program_nodes(builder=mock_builder)
+
+    assert isinstance(mock_builder.store.program, Launcher)
+
+    assert list(mock_builder.store.program._program._groups.keys()) == [
+        "data_server",
+        "parameter_server",
+        "executor",
+        "trainer",
+    ]
+    assert (
+        mock_builder.store.program._program._groups["data_server"][
+            -1
+        ]._priority_tables_fn
+        == mock_builder.data_server
+    )
+    assert (
+        mock_builder.store.program._program._groups["parameter_server"][-1]._constructor
+        == mock_builder.parameter_server
+    )
+    assert (
+        mock_builder.store.program._program._groups["executor"][-1]._constructor
+        == mock_builder.executor
+    )
+    assert (
+        mock_builder.store.program._program._groups["trainer"][-1]._constructor
+        == mock_builder.trainer
+    )
+
+    assert not "evaluator" in mock_builder.store.program._program._groups.keys()
```

Comment:
It is okay to remove this assertion since the assert on line 171 already verifies this.

Reply:
Okay, thanks for noticing it 🙌